### PR TITLE
Improve name of default value

### PR DESCRIPTION
**Pull request by som-snytt**
The odd name was copied from old method `cached`.

https://github.com/scala/scala/commit/873a28d90c6215d91bc024ba124d11030dc77596

Addresses the anomaly noticed at https://github.com/scala/bug/issues/12897

### CODE REVIEW

File: src/library/scala/collection/concurrent/Map.scala [hunk 1]
```diff
@@ -32,8 +32,6 @@ import scala.annotation.tailrec
   *  provides all of the methods a `Map` does, with the difference that all the
   *  changes are atomic. It also describes methods specific to concurrent maps.
   *
-  *  '''Note''': The concurrent maps do not accept `'''null'''` for keys or values.
-  *
```
Comment (OndrejSpanel) [inline src/library/scala/collection/concurrent/Map.scala:36]: Why is the note about non supporting null deleted? This looks sneaky, does not seem to be related to the rename.

File: src/library/scala/collection/mutable/Map.scala [hunk 2]
```diff
@@ -135,15 +135,15 @@ trait MapOps[K, V, +CC[X, Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]]
...
    *
    *  @param  key the key to test
-   *  @param  op  the computation yielding the value to associate with `key`, if
+   *  @param  defaultValue  the computation yielding the value to associate with `key`, if
    *              `key` is previously unbound.
    *  @return     the value associated with key (either previously or as a result
    *              of executing the method).
    */
-  def getOrElseUpdate(key: K, op: => V): V =
+  def getOrElseUpdate(key: K, @deprecatedName("op", since="2.13.13") defaultValue: => V): V =
```
Comment (OndrejSpanel) [inline src/library/scala/collection/mutable/Map.scala:143]: The change itself looks straightforward (almost trivial) and reasonable. I miss information (in the PR, not in the code) about how does it affect binary compatibility (both backward and forward?

File: src/library/scala/collection/concurrent/Map.scala (hunk 1 above)
Comment (som-snytt) [reply src/library/scala/collection/concurrent/Map.scala:36]: https://github.com/scala/scala/commit/606fb73b88edfc4b3ce9a68289306acc4fa8cd2b

File: src/library/scala/collection/mutable/Map.scala (hunk 2 above)
Comment (som-snytt) [reply src/library/scala/collection/mutable/Map.scala:143]: If MiMA doesn't complain, then parameter metadata via reflection is not a binary compat issue. Binary sig represents only types.

**Review by lrytz — APPROVED** [review]
👍  thank you

**Comment by som-snytt** [comment]
But this is also an odd name. Why `defaultValue` instead of `default`?

This PR aligns with `HashMap`, which called it `defaultValue`, but other methods use `default` for the same purpose, such as `getOrElse`.

Possibly, `default` is the function (`def default` and `withDefault(f)`) and `defaultValue` is a value (as in `withDefaultValue(v)`). But a better solution for uniformity would be to adjust the name of `withDefaultValue` to overload `withDefault`. The word `value` is extraneous.
